pkg/api/frontend_logging.go

```diff
@@ -28,14 +66,121 @@ type frontendSentryEvent struct {
 	Exception *frontendSentryException `json:"exception,omitempty"`
 }
 
+func guessSourceMapLocation(sourceUrl string) (*sourceMapLocation, error) {
+	u, err := url.Parse(sourceUrl)
+	if err != nil {
+		return nil, err
+	}
+	if strings.HasPrefix(u.Path, "/public/build/") {
+		return &sourceMapLocation{
+			dir:      setting.StaticRootPath,
+			path:     path.Join("build", u.Path[len("/public/build/"):]) + ".map",
+			pluginId: "",
+		}, nil
+	} else if strings.HasPrefix(u.Path, "/public/plugins/") {
+		for _, route := range plugins.StaticRoutes {
+			pluginPrefix := path.Join("/public/plugins/", route.PluginId)
+			if strings.HasPrefix(u.Path, pluginPrefix) {
+				return &sourceMapLocation{
+					dir:      route.Directory,
+					path:     u.Path[len(pluginPrefix):] + ".map",
+					pluginId: route.PluginId,
+				}, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func getSourceMap(sourceUrl string) (*sourceMap, error) {
+	sourceMapCache.Lock()
+	defer sourceMapCache.Unlock()
+
+	if smap, ok := sourceMapCache.cache[sourceUrl]; ok {
+		return smap, nil
+	}
+	sourceMapLocation, err := guessSourceMapLocation(sourceUrl)
+	if err != nil {
+		return nil, err
+	}
+	if sourceMapLocation != nil {
+		dir := http.Dir(sourceMapLocation.dir)
+		f, err := dir.Open(sourceMapLocation.path)
+		if err != nil {
+			if os.IsNotExist(err) {
+				sourceMapCache.cache[sourceUrl] = nil
+				return nil, nil
+			}
+			return nil, err
+		}
+		defer func() {
+			if err := f.Close(); err != nil {
+				frontendLogger.Error("Failed to close source map file.", "err", err)
+			}
+		}()
+		b, err := ioutil.ReadAll(f)
+		if err != nil {
+			return nil, err
+		}
+		consumer, err := sourcemap.Parse(sourceUrl+".map", b)
+		if err != nil {
+			return nil, err
+		}
+		smap := &sourceMap{
+			consumer: consumer,
+			pluginId: sourceMapLocation.pluginId,
+		}
+		sourceMapCache.cache[sourceUrl] = smap
+		return smap, nil
+	} else {
+		sourceMapCache.cache[sourceUrl] = nil
+	}
+	return nil, nil
+}
+
+func resolveSourceLocation(url string, line int, column int) (*sourceLocation, error) {
+	smap, err := getSourceMap(url)
+	if err != nil {
+		return nil, err
+	}
+	if smap != nil {
+		file, function, line, col, ok := smap.consumer.Source(line, column)
+		if ok {
+			if len(function) == 0 {
+				function = "?"
+			}
+			return &sourceLocation{
+				file:     file,
+				line:     line,
+				col:      col,
+				function: function,
+				pluginId: smap.pluginId,
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
 func (value *frontendSentryExceptionValue) FmtMessage() string {
 	return fmt.Sprintf("%s: %s", value.Type, value.Value)
 }
 
 func (value *frontendSentryExceptionValue) FmtStacktrace() string {
 	var stacktrace = value.FmtMessage()
 	for _, frame := range value.Stacktrace.Frames {
-		stacktrace += fmt.Sprintf("\n  at %s (%s:%v:%v)", frame.Function, frame.Filename, frame.Lineno, frame.Colno)
+		mappedLocation, err := resolveSourceLocation(frame.Filename, frame.Lineno, frame.Colno)
```

Comment:
Just a nit, but could you just pass on the frame? It would make for more ergonomic code, especially when the function will eventually need more or less arguments. There's no benefit to passing individual fields of `frame`.
```suggestion
		mappedLocation, err := resolveSourceLocation(frame)
```

Reply:
right! :)  
- [ ] pass the frame